clustered/client/src/main/java/org/ehcache/clustered/client/internal/config/xml/ClusteringServiceConfigurationParser.java

```diff
@@ -227,14 +251,197 @@ public URI getNamespace() {
+    }
+  }
+
+  private Element createTimeoutElement(Document doc, String timeoutName, Duration timeout) {
+    Element retElement = null;
+    if (READ_TIMEOUT_ELEMENT_NAME.equals(timeoutName)) {
+      retElement = doc.createElement(TC_CLUSTERED_NAMESPACE_PREFIX + COLON + READ_TIMEOUT_ELEMENT_NAME);
+    } else if (WRITE_TIMEOUT_ELEMENT_NAME.equals(timeoutName)) {
+      retElement = doc.createElement(TC_CLUSTERED_NAMESPACE_PREFIX + COLON + WRITE_TIMEOUT_ELEMENT_NAME);
+    } else {
+      retElement = doc.createElement(TC_CLUSTERED_NAMESPACE_PREFIX + COLON + CONNECTION_TIMEOUT_ELEMENT_NAME);
+    }
+    retElement.setAttribute(UNIT_ATTRIBUTE_NAME, DEFAULT_UNIT_ATTRIBUTE_VALUE);
+    retElement.setTextContent(Long.toString(timeout.getSeconds()));
+    return retElement;
+  }
+
+  private Element processServerSideElements(Document doc, ClusteringServiceConfiguration clusteringServiceConfiguration) {
+    Element serverSideConfigurationElem = createServerSideConfigurationElement(doc, clusteringServiceConfiguration);
+
+    if (clusteringServiceConfiguration.getServerConfiguration() != null) {
+      ServerSideConfiguration serverSideConfiguration = clusteringServiceConfiguration.getServerConfiguration();
+      String defaultServerResource = serverSideConfiguration.getDefaultServerResource();
+      if (!(defaultServerResource == null || defaultServerResource.trim().length() == 0)) {
+        Element defaultResourceElement = createDefaultServerResourceElement(doc, defaultServerResource);
+        serverSideConfigurationElem.appendChild(defaultResourceElement);
+      }
+      Map<String, Pool> resourcePools = serverSideConfiguration.getResourcePools();
+      if (!(resourcePools == null || resourcePools.isEmpty())) {
+        for (Map.Entry<String, Pool> entry : resourcePools.entrySet()) {
+          Element poolElement = createSharedPoolElement(doc, entry);
+          serverSideConfigurationElem.appendChild(poolElement);
+        }
+      }
+    }
+    return serverSideConfigurationElem;
+  }
 
+  private Element createServerSideConfigurationElement(Document doc, ClusteringServiceConfiguration clusteringServiceConfiguration) {
+    Element serverSideConfigurationElem = doc.createElement(TC_CLUSTERED_NAMESPACE_PREFIX + COLON + SERVER_SIDE_CONFIG);
+    serverSideConfigurationElem.setAttribute(AUTO_CREATE_ATTRIBUTE_NAME, clusteringServiceConfiguration.isAutoCreate()
+      ? Boolean.TRUE.toString() : Boolean.FALSE.toString());
+    return serverSideConfigurationElem;
   }
 
+
+  private Element createSharedPoolElement(Document doc, Map.Entry<String, Pool> entry) {
+    String poolName = entry.getKey();
+    Pool pool = entry.getValue();
+    Element poolElement = doc.createElement(TC_CLUSTERED_NAMESPACE_PREFIX + COLON + SHARED_POOL_ELEMENT_NAME);
+    poolElement.setAttribute(NAME_ATTRIBUTE_NAME, poolName);
+    String from = pool.getServerResource();
+    if (!(from == null || from.trim().length() == 0)) {
```

Comment:
I've seen a lot of trim. But this is Java configuration. Are we sure there can be empty space? Shouldn't then be validated when setting it on the pool in first place?

Reply:
I did not see any validation other than not null check. Making sure that empty spaces are not accidentally set.